### PR TITLE
Wallets nav2 header

### PR DESCRIPTION
**Pull request by buoyad**
- Factor `SendButton` and `DropdownButton` from the original header into wallets common components

image

r? @keybase/react-hackers

### CODE REVIEW

File: shared/constants/wallets.js [hunk 1]
```diff
@@ -532,6 +532,9 @@ export const getAccounts = (state: TypedState) => state.wallets.accountMap.value
 
 export const getSelectedAccount = (state: TypedState) => state.wallets.selectedAccount
 
+export const getSelectedAccountData = (state: TypedState) =>
+  state.wallets.accountMap.get(getSelectedAccount(state), makeAccount())
```
Comment (chrisnojima) [inline shared/constants/wallets.js]: should use unknownAccount so its not a new object every time

File: shared/wallets/common/buttons/index.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  onSendToKeybaseUser: () => void,
+  onSendToStellarAddress: () => void,
+  onSendToAnotherAccount: () => void,
+  small?: boolean,
+|}
+
+class _SendButton extends React.PureComponent<SendProps & Kb.OverlayParentProps> {
+  _menuItems = [
+    {
+      onClick: () => this.props.onSendToKeybaseUser(),
```
Comment (chrisnojima) [inline shared/wallets/common/buttons/index.js]: don't need new closures here, can just be `this.props.onSendToKeybaseUser` same below

File: shared/wallets/common/buttons/index.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  small?: boolean,
+|}
+
+class _DropdownButton extends React.PureComponent<DropdownProps & Kb.OverlayParentProps> {
+  render() {
+    const onShowSecretKey = this.props.onShowSecretKey
+    const _menuItems = [
+      onShowSecretKey
+        ? {
+            onClick: () => onShowSecretKey(),
```
Comment (chrisnojima) [inline shared/wallets/common/buttons/index.js]: same here

File: shared/wallets/common/buttons/index.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+          </Kb.Button>
+        </Kb.Box2>
+        <Kb.FloatingMenu
+          attachTo={this.props.getAttachmentRef}
+          closeOnSelect={true}
+          items={_menuItems}
+          onHidden={this.props.toggleShowingMenu}
+          visible={this.props.showingMenu}
+          position="bottom center"
+          positionFallbacks={[]}
```
Comment (chrisnojima) [inline shared/wallets/common/buttons/index.js]: we should allow this to be null so it doesn't thrash every time (can ticket)

File: shared/wallets/nav-header/add-account.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    dispatch(
+      RouteTreeGen.createNavigateAppend({path: [{props: {showOnCreation: true}, selected: 'linkExisting'}]})
+    )
+  },
+})
+
+export default Container.namedConnect<{||}, _, _, _, _>(
+  () => ({}),
+  mapDispatchToProps,
+  (_1, d, _2) => d,
```
Comment (chrisnojima) [inline shared/wallets/nav-header/add-account.js]: can just be `(_, d)`

File: shared/wallets/nav-header/container.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
+// @flow
+import * as Container from '../../util/container'
+import * as Constants from '../../constants/wallets'
+import * as Types from '../../constants/types/wallets'
+import * as RouteTreeGen from '../../actions/route-tree-gen'
+import {HeaderTitle as _HeaderTitle, HeaderRightActions as _HeaderRightActions} from '.'
+
+const mapStateToPropsHeaderTitle = state => ({
+  _account: Constants.getSelectedAccountData(state),
+  username: state.config.username || '',
```
Comment (chrisnojima) [inline shared/wallets/nav-header/container.js]: username is always a string

File: shared/wallets/common/buttons/index.js (hunk 4 above)
Comment (buoyad) [reply shared/wallets/common/buttons/index.js]: It's allowed to be undefined

File: shared/constants/wallets.js (hunk 1 above)
Comment (buoyad) [reply shared/constants/wallets.js]: ah forgot about that, I checked for `emptyAccount`

**Comment by buoyad** [comment]
Thanks @chrisnojima, made changes PTAL

File: shared/wallets/common/buttons/index.js (hunk 4 above)
Comment (buoyad) [reply shared/wallets/common/buttons/index.js]: Made desktop-9481

**Pull request merged**